sdk/servicebus/service-bus/src/core/messageSender.ts

```diff
@@ -718,12 +715,18 @@ export class MessageSender extends LinkEntity {
     });
   }
 
-  async createBatch(options?: CreateBatchOptions): Promise<ServiceBusMessageBatch> {
+  async createBatch(
+    options?: CreateBatchOptions,
+    senderOptions?: GetSenderOptions
+  ): Promise<ServiceBusMessageBatch> {
     throwErrorIfConnectionClosed(this._context.namespace);
     if (!options) {
       options = {};
     }
-    let maxMessageSize = await this.getMaxMessageSize({ retryOptions: options.retryOptions });
+    if (!senderOptions) {
```

Comment:
You know, now that we have the fancy ?. operators we probably don't need this kind of initialization anymore, right?

Reply:
Or you can give `options` and `sendOptions` both default values :)

Reply:
Defaults will be provided internally, we can keep them remained `undefined` here(in case they are `undefined`).